BookEcommerce.Models/DAL/Repositories/AuthenticationRepository.cs

```diff
@@ -46,25 +67,34 @@ public Task<IdentityResult> CreateAdmin(ApplicationUser applicationUser)
 
         public async Task<string> Login(LoginDTO LoginDTO)
         {
-            var User = await this.userManager!.FindByEmailAsync(LoginDTO.Email);
-            if (User == null) return "User does not exist";
-            if (!await this.userManager.CheckPasswordAsync(User, LoginDTO.Password)) return "Username or Password invalid";
+            try
+            {
+                var User = await this.GetUserByEmail(LoginDTO.Email!);
+                if (User == null) return null!;
+                //if (!await this.userManager.CheckPasswordAsync(User, LoginDTO.Password)) return null;
+                SignInResult result = await this.signInManager!.PasswordSignInAsync(User, LoginDTO.Password, false, false);
+                if (!result.Succeeded) return null!;
 
             List<Claim> claims = new List<Claim>
             {
+                new Claim(JwtRegisteredClaimNames.NameId, User.Id),
                 new Claim(JwtRegisteredClaimNames.Email, LoginDTO.Email!),
                 new Claim(JwtRegisteredClaimNames.Jti, Guid.NewGuid().ToString())
             };
 
-            var RoleManager = await this.userManager.GetRolesAsync(User);
+                var RoleManager = await this.userManager!.GetRolesAsync(User);
             foreach (var role in RoleManager)
             {
                 claims.Add(new Claim(ClaimTypes.Role, role));
             }
-
-            var token = tokenRepository!.CreateAccessToken(claims);
+                var token = tokenRepository!.CreateToken(claims);
             return token;
         }
+            catch (Exception e)
+            {
+                throw e;
```

Comment:
can we add logger and have error handling for this one. You catch e and throw e, it's redundant

Reply:
sure thing